src/adapters/osm_schedule.js

```diff
@@ -25,7 +25,7 @@ function scheduleStatus(scheduleResponse, timeMessages) {
 }
 
 function nextTransitionTime(seconds, nextTransitionDate) {
-  if(seconds < 12 * 60 * 60) {
+  if (nextTransitionDate && seconds < 12 * 60 * 60) {
```

Comment:
Please make explicit comparison for `nextTransitionDate` (or a comment since it seems difficult to check what's going on here).